pkg/graveler/committed/meta_range.go

```diff
@@ -26,15 +26,26 @@ type Iterator interface {
 
 // DiffIterator iterates over all Range headers and values of a Diff, allowing seeking by entire
 // ranges.
+// DiffIterator might contain ranges without headers
+// for example:
+//
+// left        [min].R1.[max]                     [min].R3.[max]        [min]...............R5..............[max]
+//        ------------------------------------------------------------------------------------------------
+// right                        [min].R2.[max]     [min.....R4....max]  [min].R6.[max] [min].R7.[max]
+//
+// R1 -  will return as diff with header
+// R2 - will return as diff with header
+// R3 and R4 - could not return a header because we must enter the ranges in order to get some header values (such as count)
+// R5 and R6 - same as R3 and R4
+// R7 - in case R5 has no values in the R7 range, R7 would return as a diff with header
```

Comment:
Neat explanation, great drawing!  I would maybe rephrase the tense to make it a bit shorter.

```suggestion
// DiffIterator may contain ranges without headers. For instance, consider the diff of
// these ranges:
//
// left        [min].R1.[max]                     [min].R3.[max]        [min]...............R5..............[max]
// ------------------------------------------------------------------------------------------------
// right                        [min].R2.[max]     [min.....R4....max]  [min].R6.[max] [min].R7.[max]
//
// R1 - returned as a diff inside range R1.
// R2 - returned as a diff inside range R2.
// R3 and R4 - not in any existing range (and creating a new range would require consuming them).
// R5 and R6 - as above, not in any existing range.
// R7 - returned as a diff inside range R7 if R5 has no keys inside the range of R7.
```